tutorials/nginx-ingress-gke/index.md

```diff
@@ -0,0 +1,353 @@
+# Ingress with Nginx controller on Google Kubernetes Engine (GKE)
+
+This guide explains how to deploy the [nginx Ingress
+Controller](https://github.com/kubernetes/ingress-nginx) on Google Kubernetes
+Engine (GKE). 
+
+In Kubernetes,
+[Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+allows external users and client applications access to HTTP services.  Ingress
+consists of two components. 
+[Ingress Resource](https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource)
+is a collection of rules for the inbound traffic to reach services.  These are
+Layer 7 (L7) rules that allow hostnames (and optionally paths) to be directed to
+specific services in Kubernetes.  The second component is the
+[Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers)
+which acts upon the rules set by the Ingress Resource, typically via an HTTP or
+L7 load balancer.  It is vital that both pieces are properly configured to route
+traffic from an  outside client to a Kubernetes service.  
+
+Nginx is a popular choice for an ingress controller for a variety of features:
+
+-  [Websocket](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/websocket),
+   which allows you to load balance Websocket applications.
+-  [SSL Services](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/ssl-services),
+   which allows you to load balance HTTPS applications.
+-  [Rewrites](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/rewrites),
+   which allows you to rewrite the URI of a request before sending it to the
+   application.
+-  [Session Persistence](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/session-persistence)
+   (NGINX Plus only), which guarantees that all the requests from the same
+   client are always passed to the same backend container.
+-  [Support for JWTs](https://github.com/nginxinc/kubernetes-ingress/blob/master/examples/jwt)
+   (NGINX Plus only), which allows NGINX Plus to authenticate requests by
+   validating JSON Web Tokens (JWTs).
+
+The following diagram shows the architecture described above:
+
+<img src="https://github.com/ameer00/community/blob/master/tutorials/nginx-ingress-gke/Nginx%20Ingress%20on%20GCP%20-%20Fig%2002.png" width="65%">
+
+This tutorial illustrates how to set up a
+[deployment in Kubernetes](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)
+with an Ingress Resource using nginx as the ingress controller to route/load
+balance traffic from external clients to the Deployment.  With this tutorial,
+admins will be able to:
+
+-  Set up an application deployment in Kubernetes
+-  Set up an Ingress Resource object for the Deployment
+-  Set up an Ingress Controller using nginx as the controller (instead of the
+   GKE default Cloud Load Balancer offered by GCP)
+
+# Objectives
```

Comment:
Adjust all headings except the title to H2 and H3 accordingly. Only the title should be H1.